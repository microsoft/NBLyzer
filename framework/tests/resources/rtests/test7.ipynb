{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from sklearn.linear_model import LinearRegression\n",
    "\n",
    "df = pd.read_csv('Whatever')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "X = df.drop('target', axis = 1)\n",
    "y = df['target']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "model = LinearRegression()\n",
    "\n",
    "def train_model(X_local_train, y_local_train, model):\n",
    "    model.fit(X_local_train, y_local_train)\n",
    "\n",
    "train_model(X, y, model)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def test_model(X_local_test, y_local_test, model):\n",
    "    model.predict(X_local_test, y_local_test)\n",
    "\n",
    "test_model(X, y, model)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.5 64-bit",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.5"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "8f91001d8d303347d46b8e57194f813c4300d3651b311e49c66d26d3b8f93dfe"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
