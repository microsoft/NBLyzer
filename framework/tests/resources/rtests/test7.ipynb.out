[{"cell_id":3,"errors":[{"line":2,"label":"X_local_test", "error_type":"ErrorType.TERMINAL", "message":"Training model with data leak."}],"path":[0, 1, 2, 3]},{"cell_id":3,"errors":[{"line":1,"label":"", "error_type":"ErrorType.TERMINAL", "message":"Idle cell."}],"path":[3]}]
